packages/xsnap/src/build.js

```diff
@@ -220,16 +220,34 @@ const updateSubmodules = async (showEnv, { env, stdout, spawn, fs }) => {
  *     existsSync: typeof import('fs').existsSync,
  *     rmdirSync: typeof import('fs').rmdirSync,
  *     readFile: typeof import('fs').promises.readFile,
+ *     writeFile: typeof import('fs').promises.writeFile,
  *   },
  *   os: {
  *     type: typeof import('os').type,
  *   }
  * }} io
+ * @param {object} [options]
+ * @param {boolean} [options.forceBuild]
  */
-const makeXsnap = async ({ spawn, fs, os }) => {
+const makeXsnap = async ({ spawn, fs, os }, { forceBuild = false } = {}) => {
   const pjson = await fs.readFile(asset('../package.json'), 'utf-8');
   const pkg = JSON.parse(pjson);
 
+  const configEnvs = [
+    `XSNAP_VERSION=${pkg.version}`,
+    `CC=cc "-D__has_builtin(x)=1"`,
+  ];
+
+  const configEnvFile = asset('../build.config.env');
+  const existingConfigEnvs = fs.existsSync(configEnvFile)
+    ? await fs.readFile(configEnvFile, 'utf-8')
+    : '';
+
+  const expectedConfigEnvs = configEnvs.concat('').join('\n');
+  if (forceBuild || existingConfigEnvs.trim() !== expectedConfigEnvs.trim()) {
+    await fs.writeFile(configEnvFile, expectedConfigEnvs);
```

Comment:
nice, so we update `build.config.env` if it differs from the `const configEnvs` here in build.js, and `build.config.env` is added as a `*.o` dependency via `EXTRA_DEPS`, so modifying most of `src/build.js` won't cause a rebuild, but modifying the `-D__has_builtin` or the `XSNAP_VERSION` value *will*.

Reply:
Or if `forceBuild` is passed, which is if we detect a version mismatch, in case someone mucked around with the binary directly